src/Notes/Note.php

```diff
@@ -503,7 +503,7 @@ public function set_date_created( $date ) {
 			$this->error( 'admin_note_invalid_data', __( 'The admin note date prop cannot be empty.', 'woocommerce-admin' ) );
 		}
 
-		$this->set_date_prop( 'date_created', $date );
+		$this->set_date_prop( 'date_created', wc_string_to_timestamp( $date ) );
```

Comment:
`wc_string_to_timestamp` says it only accepts a string. But, according to our docs for `set_date_created`, `$date` could be an integer. So, we should account for that.

Reply:
@mattsherman Nice catch! 👍 I will update the code with a test.

Reply:
@mattsherman  Fixed in https://github.com/woocommerce/woocommerce-admin/pull/6795/commits/e191299a87d7e92d2a69578c8b4b50618f961b8d 

I also created a new test `WC_Tests_Notes_Note` as we're testing behavior of `Note`